doc/lsst.pipe.base/command-line-task-data-repo-howto.rst

```diff
@@ -0,0 +1,136 @@
+.. _command-line-task-data-repo-howto:
+
+#################################################################
+Using Butler data repositories and reruns with command-line tasks
+#################################################################
+
+:ref:`Command-line tasks <using-command-line-tasks>` use Butler data repositories for reading and writing data.
+Butler data repositories are specially-configured directories on a filesystem.
+This page describes the two ways for specifying data repositories on the command line:
+
+1. The canonical method is to specify input and output repository directories with the :option:`REPOPATH` and :option:`--output` command-line arguments.
+   Read about this in :ref:`command-line-task-data-repo-using-output`.
+
+2. The alternative method is to use :option:`REPOPATH` with the :option:`--rerun` argument, which creates data repositories conveniently packaged within the root repository directory.
+   Read about this pattern in :ref:`command-line-task-data-repo-using-reruns`.
+
+.. _command-line-task-data-repo-using-output:
+
+Using input and output repositories
+===================================
+
+How to create a new output repository
+-------------------------------------
+
+To have a command-line task read data from a :file:`inputrepo` repository directory and write to a :file:`outputrepo` directory, set the :option:`REPOPATH` and :option:`--output` arguments like this:
+
+.. code-block:: text
+
+   task.py inputrepo --output outputrepo ...
+
+The :file:`outputrepo` directory will be created if it does not already exist.
+
+How to chain output repositories
+--------------------------------
+
+The output repository for one task can become the input repository for the next command-line task.
+For example:
+
+.. code-block:: text
+
+   task2.py outputrepo --output outputrepo2 ...
+
+Because Butler data repositories are *chained*, the output repository (here, :file:`outputrepo2`) will appear to contain all the datasets from the input repositories (here: :file:`inputrepo`, :file:`outputrepo`, and :file:`outputrepo2` itself).
+
```

Comment:
"will appear to contain all the datasets from the input repositories" - this statement may be misleading because data files are not actually copied/linked from `inputrepo` to `outputrepo` and on down the chain. Instead there is some behind-the-scenes configuration magic that tells each repo who its parent is.